File: umuganda/src/main/java/com/webtech/umuganda/core/attendance/service/AttendanceService.java
package com.webtech.umuganda.core.attendance.service;

import com.webtech.umuganda.core.attendance.dto.AttendanceDto;

import java.util.List;
import java.util.UUID;

public interface AttendanceService {

    AttendanceDto createAttendance(AttendanceDto dto);

    AttendanceDto updateAttendance(UUID id, AttendanceDto dto);

    AttendanceDto getAttendanceById(UUID id);

    List<AttendanceDto> getAllAttendance();

    List<AttendanceDto> getByUserId(UUID userId);

    List<AttendanceDto> getByUmugandaId(UUID umugandaId);

    void deleteAttendance(UUID id);
}

File: umuganda-frontend/src/services/attendanceService.js
import api from './api';

const attendanceService = {
    getAllAttendance: async () => {
        const response = await api.get('/attendance');
        return response.data;
    },

    getAttendanceById: async (id) => {
        const response = await api.get(`/attendance/${id}`);
        return response.data;
    },

    getByUserId: async (userId) => {
        const response = await api.get(`/attendance/user/${userId}`);
        return response.data;
    },

    getByUmugandaId: async (umugandaId) => {
        const response = await api.get(`/attendance/umuganda/${umugandaId}`);
        return response.data;
    },

    createAttendance: async (attendanceData) => {
        const response = await api.post('/attendance', attendanceData);
        return response.data;
    },

    updateAttendance: async (id, attendanceData) => {
        const response = await api.put(`/attendance/${id}`, attendanceData);
        return response.data;
    },

    deleteAttendance: async (id) => {
        await api.delete(`/attendance/${id}`);
    }
};

export default attendanceService;
